src/dishka/integrations/base.py

```diff
@@ -539,12 +574,18 @@ def wrap_injection(
     )
     get_auto_injected_func = _GET_AUTO_INJECTED_FUNC_DICT[injected_func_type]
 
-    auto_injected_func = get_auto_injected_func(
-        func=func,
-        dependencies=dependencies,
-        additional_params=additional_params,
-        container_getter=container_getter,
-    )
+    kwargs = {
+        "func": func,
+        "dependencies": dependencies,
+        "additional_params": additional_params,
+        "container_getter": container_getter,
+    }
+    if provide_context is not None:
```

Comment:
It leads that function complexity increases and ruff check throws errors - C901 and PLR0912
There was an option to pass `provide_context ` to `get_auto_injected_func` and extent all `_get_auto_injected..` functions with *args , **kwargs, but I wanted to warn user explicitly about how `provide_context` works
I guess the general refactoring of this function is needed, for example moving getting dependencies above to another function, but it's out of the PR scope, wdyt?

Reply:
```
src/dishka/integrations/base.py:502:5: C901 `wrap_injection` is too complex (12 > 10)
    |
502 | def wrap_injection(
    |     ^^^^^^^^^^^^^^ C901
503 |     *,
504 |     func: Callable[P, T],
    |

src/dishka/integrations/base.py:502:5: PLR0912 Too many branches (13 > 12)
    |
502 | def wrap_injection(
    |     ^^^^^^^^^^^^^^ PLR0912
503 |     *,
504 |     func: Callable[P, T],
```